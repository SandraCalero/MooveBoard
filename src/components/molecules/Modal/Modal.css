.glass {
	display: none;
	position: absolute;
	width: 100%;
	height: 100vh;
	justify-content: center;
	align-items: center;
	z-index: 5;
	background: #c4c4c4e8;
	top: 0;
	left: 0;
}

.glass.show {
	display: flex;
}

.modal {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	background: white;
	border: solid 1px;
	max-width: 30rem;
	height: auto;
	border: none;
	border-radius: 0.5rem;
	box-shadow: 0px 0px 10px 0px rgb(0 0 0 / 50%);
}

.modalTitle,
.modalInfo,
.divModalButtons {
	display: flex;
	align-items: center;
	padding: 1.2rem;
	margin: 0;
}

.modal .modalTitle {
	margin: 0.8rem;
}

.modal .modalInfo {
	border-top: 1px solid #c4c4c4;
	border-bottom: 1px solid #c4c4c4;
}

.modalInfo * {
	margin: 0.8rem;
}
