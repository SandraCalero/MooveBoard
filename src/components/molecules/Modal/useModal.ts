interface ModalProps {
	isModalOpen: boolean;
}

const useModal = ({ isModalOpen }: ModalProps) => {
	const flex: string = isModalOpen ? 'flex' : 'hidden';
	return { flex };
};

export default useModal;
