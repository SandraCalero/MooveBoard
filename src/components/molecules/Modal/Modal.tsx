import warning from 'assets/icons/warning.png';
import Button from 'components/atoms/Button/Button';
import './Modal.css';

interface ModalProps {
	title: string;
	message: string;
}

export default function Modal({ title, message }: ModalProps) {
	/* const wrapperClass = classNames('glass', {
		show: isConfirmationOpen,
	}); */
	return (
		<div className='glass'>
			<div className='modal'>
				<h2 className='modalTitle'>{title}</h2>
				<div className='modalMessage'>
					<img src={warning} alt='Warning icon' />
					<p>{message}</p>
				</div>

				<div className='divModalButtons'>
					<Button variant='cancelButton' text='Cancel' />
					<Button variant='confirmButton' text='Confirm' />
				</div>
			</div>
		</div>
	);
}
