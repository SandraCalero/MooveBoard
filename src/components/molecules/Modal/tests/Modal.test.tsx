import { render } from 'utils/tests/test-utils';
import Modal from '../Modal';

describe('<Modal />', () => {
	it('Snapshot', () => {
		const onConfirm = jest.fn();
		const onCancel = jest.fn();
		const { container, getByText } = render(
			<Modal
				title='Move post it to the trash'
				message='Are you sure you want to move this post it to the trash?'
				isModalOpen
				onConfirm={onConfirm}
				onCancel={onCancel}
			/>
		);
		expect(container).toMatchSnapshot();
		expect(getByText('Move post it to the trash')).toBeTruthy();
		expect(
			getByText('Are you sure you want to move this post it to the trash?')
		).toBeTruthy();
	});
});
