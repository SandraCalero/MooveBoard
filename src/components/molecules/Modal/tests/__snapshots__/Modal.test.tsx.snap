// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-center z-10 m-0 p-0 w-full
			h-full absolute inset-0 bg-transparent"
  >
    <div
      class="flex flex-col justify-center items-center bg-white max-w-xs h-auto rounded-lg shadow-2xl shadow-mine-shaft-900 dark:shadow-cerulean-700 dark:bg-cerulean-900 dark:border-none transition-colors"
    >
      <h2
        class="flex items-center p-3 m-2"
      >
        Move post it to the trash
      </h2>
      <div
        class="flex items-center p-3 m-0 border-y border-solid border-grey"
      >
        <img
          alt="Warning icon"
          class="m-2 w-10"
          src="warning.png"
        />
        <p
          class="m-2 text-sm "
        >
          Are you sure you want to move this post it to the trash?
        </p>
      </div>
      <div
        class="flex items-center p-3 m-0"
      >
        <button
          class="flex flex-col items-center justify-center border-none
			m-2 p-0	bg-transparent hover:opacity-90 active:translate-y-1 justify-center w-24 h-12 rounded-lg bg-grey dark:bg-mine-shaft-500 dark:text-white transition-colors"
          name="button"
          type="button"
        >
          Cancel
        </button>
        <button
          class="flex flex-col items-center justify-center border-none
			m-2 p-0	bg-transparent hover:opacity-90 active:translate-y-1 justify-center w-24 h-12 rounded-lg bg-blue text-white dark:text-blue dark:bg-white font-semibold transition-colors"
          name="button"
          type="button"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
